src/libraries/Common/tests/System/Collections/IDictionary.Generic.Tests.cs

```diff
@@ -289,7 +289,7 @@ public void IDictionary_Generic_ItemGet_MissingNonDefaultKey_ThrowsKeyNotFoundEx
         [MemberData(nameof(ValidCollectionSizes))]
         public void IDictionary_Generic_ItemGet_MissingDefaultKey_ThrowsKeyNotFoundException(int count)
         {
-            if (DefaultValueAllowed)
+            if (DefaultValueAllowed && !IsReadOnly)
```

Comment:
this test was working only because existing int-based tests  did not generate 0 as a key:

https://github.com/dotnet/runtime/blob/c06d77a74f64d6c5b0280db56fb0c6a43c0aa907/src/libraries/Common/tests/System/Collections/IDictionary.Generic.Tests.cs#L295-L297